File: backend/src/main/java/com/anmory/yunji/entity/Article.java
package com.anmory.yunji.entity;

import lombok.Data;
import java.time.LocalDateTime;

/**
 * 孕期科普文章
 * 对应 article 表
 */
@Data
public class Article {
    private Integer articleId;
    private String title;
    private String summary;
    private String content;
    private String coverUrl;
    private String category;
    private Integer sortOrder;
    private Boolean isPublished;
    /** 受众：all=全部, pregnant=孕妇, spouse=配偶 */
    private String audience = "all";
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/lib/api/article.ts
import { apiGet } from "@/lib/api"

export interface Article {
  articleId: number
  title: string
  summary?: string
  content?: string
  coverUrl?: string
  category?: string
  sortOrder?: number
  isPublished?: boolean
  createdAt?: string
  updatedAt?: string
}

export function listArticles(userId?: number) {
  const params: Record<string, string> = {}
  if (userId != null) params.userId = String(userId)
  return apiGet<Article[]>("/api/article/list", Object.keys(params).length ? params : undefined)
}

export function getArticle(articleId: number) {
  return apiGet<Article>(`/api/article/${articleId}`)
}
